packages/smooth_app/lib/pages/home_page.dart

```diff
@@ -80,279 +80,36 @@ class _HomePageState extends State<HomePage> {
   }
 }
 
-// Note: In addition to the history of scanned products, this page currently
-// contains the user's lists and other stuff from the old Smoothie home page.
-// The intention is to move those features to other screens later or remove them
-// altogether.
-class OldHomePage extends StatefulWidget {
-  const OldHomePage();
-
+class HistoryPage extends StatefulWidget {
+  const HistoryPage();
   @override
-  State<OldHomePage> createState() => _OldHomePageState();
+  State<HistoryPage> createState() => _HistoryPageState();
 }
 
-class _OldHomePageState extends State<OldHomePage> {
-  static const ColorDestination _COLOR_DESTINATION_FOR_ICON =
-      ColorDestination.SURFACE_FOREGROUND;
-  static const Icon _ICON_ARROW_FORWARD = Icon(Icons.arrow_forward);
-
-  late DaoProductList _daoProductList;
+class _HistoryPageState extends State<HistoryPage> {
+  late DaoProductList _daoProductList ;
+  final productList = ProductList(
+    listType: ProductList.LIST_TYPE_HISTORY,
+    parameters: '',
+  );
+  var loadedList = false;
 
   @override
   Widget build(BuildContext context) {
     final LocalDatabase localDatabase = context.watch<LocalDatabase>();
     _daoProductList = DaoProductList(localDatabase);
-    final ThemeData themeData = Theme.of(context);
-    final ColorScheme colorScheme = themeData.colorScheme;
+    if (!loadedList) {
+      loadedList = true;
+      _daoProductList.get(productList).then((_) {
+       setState(() {});
+      });
```

Comment:
todo to handle error case?

Reply:
Do you have a reference for someplace in the code where the a dao fetch fails, so that I can imitate? The code I refactored to get to this state didn't have any error handling :/

Reply:
I found a poor example in `database_product_list_supplier`, but feel free to add a TODO because we need to handle these errors everywhere, with perhaps a couple retries, right now there isn't a retry, I asked Kate for advice on how we can do that. And will create a github issue for handling all error cases. TODO will make it easier to locate.

Reply:
After looking through the code, I get the feeling that there is no error handling since this call is to a local database using a known-to-work query, and nobody would know how to handle an exception caused by that.

I would guess `database_product_list_supplier` doesn't catch errors in `DaoProductList.get(..)` but (I'm guessing) in `ProductQuery.getProductList()`.

WDYT?

Reply:
yeah, on second look it seems like `database_product_list_supplier` doesn't catch errors in DaoProductList either.

If it's a known-to-work query, then it does sound good to ignore the error case.

Reply:
:+1: